cpp/FbxCliWrapper/FBXMeshData.cpp

```diff
@@ -5,41 +5,71 @@ namespace FbxClrWrapper
 {
 	FBXMeshDataInternal::FBXMeshDataInternal(FBXMeshData^ SrcData)
 	{
-		mIndices.resize(SrcData->mInds->Count);
-		mVertices.resize(SrcData->mVerts->Count);
-		mFaceSize.resize(SrcData->mFcSz->Count);
+		auto indicesArray = SrcData->mIndicesAttribute->ToInts();
+		auto verticesArray = SrcData->mVerticesAttribute->ToFloats();
+		auto faceSizeArray = SrcData->mFaceSizeAttribute->ToInts();
 
-		for (int i = 0; i < SrcData->mInds->Count; i++)
+		mIndices.resize(indicesArray->Count);
+		mVertices.resize(verticesArray->Count);
+		mFaceSize.resize(faceSizeArray->Count);
+
+		for (int i = 0; i < indicesArray->Count; i++)
 		{
-			mIndices[i] = SrcData->mInds[i];
+			mIndices[i] = indicesArray[i];
 		}
-		for (int i = 0; i < SrcData->mVerts->Count; i++)
+		for (int i = 0; i < verticesArray->Count; i++)
 		{
-			mVertices[i] = SrcData->mVerts[i];
+			mVertices[i] = verticesArray[i];
 		}
-		for (int i = 0; i < SrcData->mFcSz->Count; i++)
+		for (int i = 0; i < faceSizeArray->Count; i++)
 		{
-			mFaceSize[i] = SrcData->mFcSz[i];
+			mFaceSize[i] = faceSizeArray[i];
 		}
 	}
 
-	FBXMeshData::FBXMeshData(FBXMeshDataInternal& SrcData)
+	template <typename tData>
+	ref class ArrayDataLookup
 	{
-		mIndices = gcnew array<int32_t>(SrcData.mIndices.size());
-		mVertices = gcnew array<float>(SrcData.mVertices.size());
-		mFaceSize = gcnew array<int>(SrcData.mFaceSize.size());
-		for (size_t i = 0; i < SrcData.mIndices.size(); i++)
-		{
-			mIndices[i] = SrcData.mIndices[i];
-		}
-		for (size_t i = 0; i < SrcData.mVertices.size(); i++)
-		{
-			mVertices[i] = SrcData.mVertices[i];
-		}
-		for (size_t i = 0; i < SrcData.mFaceSize.size(); i++)
+	public:
+		std::vector<tData>& mArray;
+
+		ArrayDataLookup(std::vector<tData>& Array) :
+			mArray(Array)
+		{}
+
+		tData Lookup(int n)
 		{
-			mFaceSize[i] = SrcData.mFaceSize[i];
+			return mArray[n];
 		}
+	};
+
+	template <typename tData>

```

Comment:
I suggest that all template parameters should start with a capital T, and if there is only one, just named it T. Generally it is  well understood that the T implies a type parameter.